documentation/source/protocol_extension.rst

```diff
@@ -0,0 +1,90 @@
+How to write a protocol extension
+=================================
+
+Cocotb take care of the core communication between the simulator and testbench.
+To use some known busses like wishbone, avalon, spi, i2c,... An extension is required.
+
+This is a guideline that will explain how to write a python module protocol
+extension for cocotb.
+
+Naming convention
+-----------------
+
+A cocotb extension module must begin by `cocotb_` and terminate with protocol
+name.
+
+For exemple, with spi the module extension is `cocotb_spi`.
+
+Bus driver
+----------
+
+A Bus driver is a python class that factorize common behavior and check for
+presence of some signals.
+
+A Bus driver is an active class that be driven in the main testbench. Example is
+given with avalon bus in `cocotb/drivers/avalon.py`.
+
+With AvalonMaster class we can generate stimulis for reading and writing on bus.
+
+The main class is named `BusDriver` and is defined in `cocotb/drivers/__init__.py`
+
+TODO: give an example.
+
+Bus Monitor
+-----------
+
+A Bus monitor is mainly a passive class that will monitor the bus.
+
+TODO: explain monitor
+
+Python module
+-------------
+
+The module file hierarchy should be as following ::
```

Comment:
I think we should link to the pypa distribution docs here, rather than repeating them